### PR TITLE
Add support for Celsia CZC1

### PR DESCRIPTION
**Pull request by libanp**
Initial working code

### CODE REVIEW

**Comment by libanp** [comment]
Follows discussion at #2389

This decoder works, so I thought I'd start the PR with what I've got. Aside from whatever else comes up, there are two issues with it I'd appreciate direction for:

- Are there better utility functions for converting the raw bits to symbols? I didn't have much luck finding them
- My use of a 'type' json field doesn't fit with fields described in [DATA_FORMAT.md](https://github.com/merbanan/rtl_433/blob/master/docs/DATA_FORMAT.md), should I be using 'subtype' instead?

File: src/devices/celsia_czc1.c [hunk 1]
```diff
@@ -0,0 +1,177 @@
+/** @file
+    Bresser Weather Center 5-in-1.
```
Comment (zuckschwerdt) [inline src/devices/celsia_czc1.c:2]: "Celsia CZC1 Thermostat." ;)

File: src/devices/celsia_czc1.c [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+*/
+
+#include "bitbuffer.h"
+#include "decoder.h"
+#include "decoder_util.h"
+#include "r_device.h"
+#include <bits/stdint-uintn.h>
+
+/** @fn int celsia_czc1_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+Celsia CZC1 Thermostat
```
Comment (zuckschwerdt) [inline src/devices/celsia_czc1.c:19]: Add a dot `.` at the end

File: src/devices/celsia_czc1.c [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+A PID thermostat compatible with various manufacturers' heaters.
+
+demod: OOK_PCM
+short: 1220
+long: 1220
+reset: 4880
+
+A packet starts with a sync of {40}cccccccccccccccccccc, followed by a preamble
```
Comment (zuckschwerdt) [inline src/devices/celsia_czc1.c:28]: cc... is the preamble, 55.. is the sync

File: src/devices/celsia_czc1.c [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+        return DECODE_FAIL_MIC;
+    }
+    int id = (b[0] << 8) | b[1];
+
+    data_t *data;
+    if (b[2] == 0x00) { // pair packet
+        /* clang-format off */
+        data = data_make(
+            "model",    "",               DATA_STRING, "Celsia CZC1",
+            "id",       "ID",             DATA_FORMAT, "0x%X", DATA_INT, id,
```
Comment (zuckschwerdt) [inline src/devices/celsia_czc1.c:129]: use `"%x"` please

**Comment by zuckschwerdt** [comment]
Remove the "type" key you have. There is no need for that info, it's either packets with "heat" or empty packets.

I can rewrite the bit decoder later, basically I would use the fact that when aligned a byte will hold exactly 2 bits, i.e. it can only ever be 55, 5a, a5, aa. Otherwise using `extract_bits_symbols()` will also work https://github.com/merbanan/rtl_433/blob/master/include/util.h#L84

File: src/devices/celsia_czc1.c (hunk 1 above)
Comment (libanp) [reply src/devices/celsia_czc1.c]: Doh!

**Comment by libanp** [comment]
I've updated the code with the corrections. The decoder matches against byte values now, but I'm happy for the it to be rewritten :)

File: src/devices/celsia_czc1.c [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+    Copyright (C) 2023 Liban Hannan <liban.p@gmail.com>
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+*/
+
+#include "bitbuffer.h"
+#include "decoder.h"
```
Comment (zuckschwerdt) [inline src/devices/celsia_czc1.c:13]: You should only need "decoder.h"

**Comment by zuckschwerdt** [comment]
Looks good. In the long term we might want to put the bits into a `uint8_t[]` instead of the (large) bitbuffer_t. But that's an optimization for later (e.g. manchester_decode() also needs that refactor some day).

**Pull request merged**